File: smallpict-php/src/Exceptions/NetworkException.php
<?php

declare(strict_types=1);

namespace SmallPict\Exceptions;

class NetworkException extends SmallPictException
{
    /**
     * @param string $message
     * @param array<string, mixed> $details
     * @param \Throwable|null $previous
     */
    public function __construct(
        string $message = 'Network communication failed',
        array $details = [],
        ?\Throwable $previous = null
    ) {
        parent::__construct("Network communication failed: {$message}", 'NETWORK_ERROR', 0, null, $details, $previous);
    }
}

File: smallpict-java/src/main/java/com/smallpict/errors/NetworkException.java
package com.smallpict.errors;

public class NetworkException extends SmallPictException {
    public NetworkException(String message) {
        super(message, "NETWORK_ERROR", null, null, null, null);
    }

    public NetworkException(String message, Throwable cause) {
        super(message, "NETWORK_ERROR", null, null, null, cause);
    }
}
